Assets/Scripts/AbyssCollider.cs

```diff
@@ -0,0 +1,40 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class AbyssCollider : MonoBehaviour {
+
+    private const string LIVES_GAME_OBJ_NAME = "Remaining Lives";
+    [SerializeField]
+    private PlayerHealthManager playerHealthManager;
+    [SerializeField]
+    private LevelManager levelManager;
+    [SerializeField]
+    private PlayerMovement player;
+
+    void OnTriggerExit2D(Collider2D coll)
+    {
+        GameObject gObj = coll.gameObject;
+        if (gObj == player.gameObject)
+        {
+            playerHealthManager.LoseLife();
+            if(playerHealthManager.IsAlive()){
+                player.Reset();
+                UpdateLivesComponent(playerHealthManager.RemainingLives);
+            }
+        }
+    }
+
+    private void UpdateLivesComponent(int remainingLives)
+    {
+        char fontAwesomeHealthSymbol = '';
```

Comment:
I would think that GameObject comparison is going to be heavier than looking at the tag on the coll.gameObject?